includes/compat/subscriptions/class-wc-payment-gateway-wcpay-subscriptions-compat.php

```diff
@@ -380,4 +384,89 @@ public function render_custom_payment_meta_input( $subscription, $field_id, $fie
 		}
 		echo '</select>';
 	}
+
+	/**
+	 * Add specific data like last 4 digit of wcpay payment gateway
+	 *
+	 * @param string          $old_payment_method_title Payment method title, eg: Credit card.
+	 * @param string          $old_payment_method Payment gateway id.
+	 * @param WC_Subscription $subscription The subscription order.
+	 * @return string
+	 */
+	public function get_specific_old_payment_method_title( $old_payment_method_title, $old_payment_method, $subscription ) {
+		// make sure payment method is wcpay's.
+		if ( WC_Payment_Gateway_WCPay::GATEWAY_ID !== $old_payment_method ) {
+			return $old_payment_method_title;
+		}
+
+		$new_payment_method = $subscription->get_payment_method();
+
+		// if old and new payment method are different, no need to show specific.
+		if ( $old_payment_method !== $new_payment_method ) {
+			return $old_payment_method_title;
+		}
+
+		$last_order_id = $subscription->get_last_order();
+		if ( ! $last_order_id ) {
+			return $old_payment_method_title;
+		}
+
+		$last_order = wc_get_order( $last_order_id );
```

Comment:
I've tested things locally and see that new tokens are added to the subscription itself rather that to related orders. In this case taken token from last order might be incorrect.

Reply:
Your comment here actually made me realized that I didn't think of the case where payment is changed on the subscriptions order. I tested payment change for the renewal order only. I will put more lengthy explanation below, also for knowledge transfer about this PR to @dmvrtx.

Reply:
I reviewed the flows and indeed found one branch that was not properly tested. I updated the test suite to trigger that branch and everything looks well.